core/src/main/java/org/kohsuke/stapler/Function.java

```diff
@@ -322,15 +329,24 @@ protected MethodHandle handle() {
             return handle;
         }
 
+        @Override
         public Object invoke(StaplerRequest req, StaplerResponse rsp, Object o, Object... args) throws IllegalAccessException, InvocationTargetException {
-            Object[] arguments = new Object[args.length + 1];
-            arguments[0] = o;
-            System.arraycopy(args, 0, arguments, 1, args.length);
+            Object[] arguments;
+            if (Modifier.isStatic(m.getModifiers())) {
```

Comment:
Note that `Method.invoke` accepts a non-null first argument for a `static` method (it is just ignored).